cpu/stm32_common/include/periph_cpu_common.h

```diff
@@ -198,15 +198,25 @@ typedef struct {
     uint8_t chan;           /**< DAC device used for this line */
 } dac_conf_t;
 
+
+/**
+ * @brief   Timer type
+ */
+typedef enum {
+    TIMER_TYPE_GPTIM, /**< General-purpose timer */
+    TIMER_TYPE_LPTIM, /**< Low-power timer */
+} timer_type_t;
+
 /**
  * @brief   Timer configuration
  */
 typedef struct {
-    TIM_TypeDef *dev;       /**< timer device */
+    TIM_TypeDef *dev;       /**< timer device, not set for LPTIM */
```

Comment:
Main reason is that `LPTIM1` is not a `TIM_TypeDef *` so we could either use an union or a `void *`.

We would then need a `dev_lptim(tim)` to return the pointer with the right type.